Assets/Weapons/FireWeapon.cs

```diff
@@ -0,0 +1,334 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+public class FireWeapon : Weapon
+{
+
+    [Header("Fire Weapon")] 
+    public FireWeaponData fireWeaponData;
+
+    //Gun muzzle wil be used for FX
+    public Transform gunMuzzle;
+    //The transform where the ray casts will be emitted
+    public Transform dispersionTransform;
+
+    private int _currentAmmo;
+    private bool _isReloading;
+
+    //Fire rate global variables
+    private float _fireRateTimer;
+    private float _fireRateCurveTimer;
```

Comment:
Is having two timers for the some thing really necessary?

Reply:
Yes, because one is the global curve timer and the other one is the timer between shots